frontend/src/api/k8s/storageClasses.ts

```diff
@@ -1,9 +1,57 @@
-import { k8sListResource } from '@openshift/dynamic-plugin-sdk-utils';
-import { StorageClassKind } from '~/k8sTypes';
+import {
+  k8sGetResource,
+  k8sListResource,
+  k8sPatchResource,
+  Patch,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import { K8sAPIOptions, StorageClassConfig, StorageClassKind } from '~/k8sTypes';
 import { StorageClassModel } from '~/api/models';
+import { applyK8sAPIOptions } from '~/api/apiMergeUtils';
 
 export const getStorageClasses = (): Promise<StorageClassKind[]> =>
   k8sListResource<StorageClassKind>({
     model: StorageClassModel,
     queryOptions: {},
   }).then((listResource) => listResource.items);
+
+export const getStorageClass = (name: string): Promise<StorageClassKind> =>
+  k8sGetResource<StorageClassKind>({
+    model: StorageClassModel,
+    queryOptions: { name },
+  });
+
+export const updateStorageClassConfig = async (
+  name: string,
+  config: Partial<Omit<StorageClassConfig, 'lastModified'>> | undefined,
+  opts?: K8sAPIOptions,
+): Promise<StorageClassConfig> => {
+  const oldStorageClassResource = await getStorageClass(name);
+  const patches: Patch[] = [
+    {
+      op: 'replace',
+      path: '/metadata/annotations/opendatahub.io~1sc-config',
+      value: config
+        ? JSON.stringify({
+            ...(oldStorageClassResource.metadata.annotations?.['opendatahub.io/sc-config'] &&
```

Comment:
Nit: For all the 'opendatahub.io/sc-config', we can replace it with the type [here](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/k8sTypes.ts#L64)